src/reflect/scala/reflect/runtime/JavaMirrors.scala

```diff
@@ -949,18 +949,16 @@ private[scala] trait JavaMirrors extends internal.SymbolTable with api.JavaUnive
         if (split > 0) packageNameToScala(fullname take split) else this.RootPackage
       val owner = ownerModule.moduleClass
       val name = TermName(fullname) drop split + 1
-      val opkg = owner.info decl name
-      if (opkg.hasPackageFlag)
-        opkg.asModule
-      else if (opkg == NoSymbol) {
+      val opkg = owner.info declPackage name
```

Comment:
I'd prefer `owner.info.decl(name).filter(_.hasFlag(PACKAGE))` here rather than adding the specialized `declPackage`.

Reply:
As I understand it, the original search method (findDecl) contains a combination of symbol traversing, filtering and overriding. I just increased the abstraction of the filtering method in the "findDeclPredecate" method, adding a filter function (condition parameter) there instead of checking for the symbol flags. If the symbol is not excluded by the filter in line 955 (on the edited version of the Types.scala), then the override principle in line 1005 will be applied further to it.
 "baseClasses.head.newOverloaded(this, alts)"
This can lead to an undesirable side effect in our case.